### PR TITLE
Add new Bourse Direct 2FA capacities

### PR DESCRIPTION
**Pull request by hlestrat**
Bourse Direct has introduce 2FA systems on their web sites.

### CODE REVIEW

File: entries/b/boursedirect.fr.json [hunk 1]
```diff
@@ -2,10 +2,8 @@
     "Bourse Direct": {
         "domain": "boursedirect.fr",
         "url": "https://www.boursedirect.fr",
-        "contact": {
-            "twitter": "boursedirect",
-            "facebook": "Boursedirect"
-        },
+        "tfa": ["sms", "totp"],
```
Comment (hkamran80) [inline entries/b/boursedirect.fr.json]:
```suggestion
        "tfa": [
            "sms",
            "totp"
        ],
```

**Review by hkamran80 — CHANGES_REQUESTED** [review]

**Comment by hkamran80** [comment]
Would this count as [authentication or authorization](https://github.com/2factorauth/twofactorauth/blob/master/CONTRIBUTING.md#authorization)?
> Passwords alone are no longer enough to effectively protect your account.
You will be asked for this reinforced authentication regularly for access to your customer area and for all your “sensitive” operations.

>  **When will I be asked for reinforced authentication?**
> - Every 30 days for access to your account (in addition to your password).
> - When you log in on a new device or browser.
> - When you carry out a so-called "sensitive" operation such as: a change of password, a connection to a new device, an outgoing transfer, an IBAN addition, etc.

**Comment by hlestrat** [comment]
> Would this count as [authentication or authorization](https://github.com/2factorauth/twofactorauth/blob/master/CONTRIBUTING.md#authorization)?
> 
> > Passwords alone are no longer enough to effectively protect your account.
> > You will be asked for this reinforced authentication regularly for access to your customer area and for all your “sensitive” operations.
> 
> > **When will I be asked for reinforced authentication?**
> > 
> > * Every 30 days for access to your account (in addition to your password).
> > * When you log in on a new device or browser.
> > * When you carry out a so-called "sensitive" operation such as: a change of password, a connection to a new device, an outgoing transfer, an IBAN addition, etc.

I think both. They force to use 2FA on authentication, and they force to use one to validate some sensitive operations.

**Comment by hkamran80** [comment]
> I think both. They force to use 2FA on authentication, and they force to use one to validate some sensitive operations.

Do they require 2FA on every login or only on attempts from new devices/after 30 days?

**Comment by hlestrat** [comment]
> > I think both. They force to use 2FA on authentication, and they force to use one to validate some sensitive operations.
> 
> Do they require 2FA on every login or only on attempts from new devices/after 30 days?

You can choose between 30 days on a "confidence" browser, or every time. But on every sensitive action, you will be asked for the 2FA whether or not you asked for 30 days on login.

**Review by hkamran80 — APPROVED** [review]

**Pull request merged**